python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -286,6 +268,25 @@ def decorated(self, *args, **kwargs):
             return f(self, *args, **kwargs)
     return decorated
 
+def global_disk_lock(f):
+    @functools.wraps(f)
+    def decorated_global(self, *args, **kwds):
+        with global_disk_lock.global_lock:
+            while True:
+                try:
+                    self.global_lock_acquire()
```

Comment:
This is not intended to have a timeout. If you can't access the global file lock you can't do anything. This doesn't lead to deadlocks as the global lock (within 1 session) is released correctly once it's no longer needed. Adding a timeout here caused errors to be thrown when a thread can't access the global file lock vs the thread waiting for access which is what you'd expect.